libs/shared/components/src/top-nav/ConnectedButton.tsx

```diff
@@ -57,7 +58,7 @@ export function ConnectedButton({
             height: (theme) => theme.spacing(3),
           }}
         />
-        <UserId userId={userId} />
+        <MiddleTruncated>{userId}</MiddleTruncated>
```

Comment:
Should this be a component or just a function that acts on string?

Reply:
it needs to be a component because you don't know in advance where the ellipsis will happen. This component allows for having a "resizable" label that always keep the `end` digits visible